### PR TITLE
Bugfix and Support time attribute

### PR DESCRIPTION
**Pull request by myuhe**
Bug fix for difference of timezone
 Support starttime and duetime

### CODE REVIEW

File: org-toodledo.el [hunk 1]
```diff
@@ -1105,7 +1109,15 @@ an alist of the task fields."
           (aput 'info "goal" (org-toodledo-goal-to-id (org-entry-get nil "Goal"))))
         
         (when deadline
-          (aput 'info "duedate" (format "%.0f" (org-time-string-to-seconds deadline)))
+          ;;(aput 'info "duedate" (format "%.0f" (org-time-string-to-seconds deadline)))
+          (aput 'info "duedate" (format "%.0f" 
+                                        (+ (car (current-time-zone)) (org-time-string-to-seconds deadline))))
```
Comment (christopherjwhite) [inline org-toodledo.el:1114]: I don't understand why the time-zone offset needs to be added.  The "deadline" string doesn't have a timezone specifier and by my testing, org-time-string-to-seconds assumes local time, thus handles the offset automatically.  I grabbed these set of lines into a small lisp test as follows:

(let (info
      (deadline "<2012-01-29 Thu 09:00>"))

  (aput 'info "duedate" (format "%.0f" 
                                (+ (car (current-time-zone)) (org-time-string-to-seconds deadline))))
  (unless 
      (and
       (= 0 (caddr (org-parse-time-string deadline)))
       (= 0 (cadr (org-parse-time-string deadline))))
    (aput 'info "duetime" (format 
                           "%.0f" (+ (car (current-time-zone)) (org-time-string-to-seconds deadline)))))

  (format "\ndeadline: %s\nduetime:  %s\nduedate:  %s\ncurzone:  %s"
          deadline
          (format-time-string "%Y-%m-%d %H:%M:%S %z" (seconds-to-time (string-to-number (cdr (assoc "duetime" info)))))
          (format-time-string "%Y-%m-%d %H:%M:%S %z" (seconds-to-time (string-to-number (cdr (assoc "duedate" info)))))
          (current-time-zone)
          )
  )
"
deadline: <2012-01-29 Thu 09:00>
duetime:  2012-01-29 04:00:00 -0500
duedate:  2012-01-29 04:00:00 -0500
curzone:  (-18000 -0500)"

By adding the offset (-5h), the duetime/duedate are getting shifted backwards relative to my current time.

Comment (myuhe) [inline org-toodledo.el:1114]: Toodledo has timezone.
http://www.toodledo.com/account_edit.php?edit=4

So, I think timezone should be added.

In you code, pushing to Toodledo , Toodledo gets invalid duetime.

Comment (christopherjwhite) [inline org-toodledo.el:1114]: Ok, so I spent some time this morning testing out the "duetime" field.
I believe there is a bug with Toodledo.com and proper handling of
timezones.  I have submitted a post to the forums on toodledo.com to
get some answers.

  https://www.toodledo.com/forums/7/13864/0/timezone-problems-with-duetime.html

They usually seem to be pretty responsive, so I'll hold off on any
changes for a few days to see what they come back with.  Hopefully it
won't be long and then I can add support into org-toodledo for the
duetime/starttime fields.

Thanks!
...cj

On Sat, 28 Jan 2012 14:48:25 -0800
myuhe
reply@reply.github.com
wrote:

**Comment by christopherjwhite** [comment]
I added support for duetime/starttime in the latest release.  I did _not_ include the timezone adjustment that you have in your code.  I'm waiting to see what the folks at toodledo.com have to say about things, as I think it's really not handled properly on the server.  The current behavior is that if you put a time in via org-mode in local time, it will show up as offset on toodledo.com and vice-versa.

**Pull request merged**